backend/app/routes/web/search_suggestions.py

```diff
@@ -0,0 +1,112 @@
+"""The UI Search endpoint"""
+import asyncio
+import logging
+from json import JSONDecodeError
+
+from fastapi import APIRouter, Depends, HTTPException, Query
+from httpx import AsyncClient, TransportError
+from pydantic.typing import Literal
+from requests import Response
+
+from app.solr.operations import search_dep
+
+from ..util import DEFAULT_SORT
+
+router = APIRouter()
+
+logger = logging.getLogger(__name__)
+
+
+SortUi = Literal["dmr", "dlr", "mp", "r", "default"]
+
+
+# pylint: disable=too-many-arguments
+@router.post("/search-suggestions", name="web:post-search-suggestions")
+async def search_suggestions(
+    collection: str = Query(..., description="Collection"),
+    q: str = Query(..., description="Free-form query string"),
+    qf: str = Query(..., description="Query fields"),
+    fq: list[str] = Query(
+        [],
+        description="Filter query",
+        example=["journal:Geonomos", 'journal:"Solar Energy"'],
+    ),
+    results_per_collection: int = Query(
+        3, description="Row count per collection", gte=3, lt=10
+    ),
+    search=Depends(search_dep),
+):
+    """
+    Main function performing the search for suggestions.
+    Serves as the dispatch function for `all_collection` search request.
+    The q, qf, fq, sort params correspond to
+    https://solr.apache.org/guide/8_11/query-syntax-and-parsing.html.
+    """
+    all_collection = (
+        "publication",
+        "dataset",
+        "software",
+        "service",
+        "data_source",
+        "training",
+        "guideline",
+        "bundle",
+    )
+
+    collections = all_collection if "all_collection" in collection else (collection,)
+    gathered_result = await asyncio.gather(
+        *[
+            _search(col, q, qf, fq, results_per_collection, search)
+            for col in collections
+        ]
+    )
+
+    return dict(gathered_result)
+
+
+async def _search(
+    collection: str = Query(..., description="Collection"),
+    q: str = Query(..., description="Free-form query string"),
+    qf: str = Query(..., description="Query fields"),
+    fq: list[str] = Query(
+        [],
+        description="Filter query",
+        example=["journal:Geonomos", 'journal:"Solar Energy"'],
+    ),
+    results_per_collection: int = Query(
+        3, description="Row count per collection", gte=3, lt=10
+    ),
+    search=Depends(search_dep),
+):
```

Comment:
It would be good to type return type, especially that this function returns tuple, which can be pretty tricky sometimes.